src/main/groovy/life/qbic/controller/SamplesController.groovy

```diff
@@ -111,14 +122,11 @@ class SamplesController {
     if(!RegExValidator.isValidSampleCode(sampleId)) {
       return HttpResponse.status(HttpStatus.BAD_REQUEST, "Not a valid sample code!")
     }
-    boolean found = sampleService.searchSample(sampleId)!=null;
-    if(found) {
-      sampleService.updateSampleStatus(sampleId, status);
-      String msg = "Sample status updated."
-      HttpResponse response = HttpResponse.status(HttpStatus.CREATED, msg)
-      response.body(msg)
-      return response
-    } else {
+    if(null != sampleService.searchSample(sampleId)){
+      sampleService.updateSampleStatus(sampleId, status)
+      return HttpResponse.status(HttpStatus.CREATED, "Sample status updated.")
```

Comment:
I am not sure what can go wrong here, but maybe we should create a case for this. E.g. catch an exception and return 500 (internal server error)?

Reply:
to elaborate: the DBManager can throw an error here that does not have a corresponding HttpResponse

Reply:
yes, this exception needs to be handled